waspc/src/Wasp/AppSpec/Valid.hs

```diff
@@ -379,6 +396,42 @@ validateWebAppBaseDir spec = case maybeBaseDir of
     startsWithSlash ('/' : _) = True
     startsWithSlash _ = False
 
+validateUserNodeVersionRange :: AppSpec -> [ValidationError]
+validateUserNodeVersionRange spec =
+  concat
+    [ checkUserRangeIsInWaspRange,
+      checkUserRangeAllowsOnlyMinorChanges
+    ]
+  where
+    userRange = AS.userNodeVersionRange spec
+    userVersionInterval = SVB.versionBounds userRange
+    waspVersionInterval = SVB.versionBounds $ SV.backwardsCompatibleWith oldestWaspSupportedNodeVersion
+    checkUserRangeIsInWaspRange :: [ValidationError]
+    checkUserRangeIsInWaspRange =
+      if not (userVersionInterval `SVB.isSubintervalOf` waspVersionInterval)
+        then
+          [ GenericValidationError $
+              "Node version range that you specified for your Wasp app ("
+                <> show userRange
+                <> ") has too low lower boundary. It must be >= "
+                <> show oldestWaspSupportedNodeVersion
+          ]
+        else []
+    checkUserRangeAllowsOnlyMinorChanges :: [ValidationError]
+    checkUserRangeAllowsOnlyMinorChanges =
+      maybe
+        [ GenericValidationWarning $
```

Comment:
@infomiho because when developing an app, you want to have dev environment same as your deployment env, to avoid any inconsistencies. When developing a library, it is different, you need to support multiple dev envs.

I don't know how to easily add that into this comment hm, but I do agree it would be good give them more info.
How about "... which is not advised. When developing an app you want to keep dev environment consistent to avoid any surprises."?

@sodic would you care providing any suggestions? If I had ideas how to make it less verbose, I would have already done so. It actually seems to be evolving in the opposite direction, we are going to be making it longer (which makes sense).

Reply:
I'm not against making it longer.
By "less verbose", I meant "simpler language and a better information to noise ratio."

Possible improvements for the current sentence:
- Removing the passive voice.
- A more conventional sentence structure.
- _major_ -> _breaking_ - Many users don't know semantic versioning but do know what a breaking change is.
- Removing unnecessary words.

Here's how the result looks like.

Instead of "Node version range that you specified for your Wasp app (range) allows major version changes, which is not advised," I suggest "Your app's Wasp version range (range) allows breaking changes . To ensure consistency across environments, we recommend keeping the version range as narrow as possible."

Reply:
Ok, let's work with this!

It is not "Wasp version range" it is "Node version range" however. Let's see if I can work that in:
```
Your app's node version range (<range>) allows breaking changes. To ensure consistency between development and production environments, we recommend you narrow down your node version range to not allow breaking changes.
```

This also tells them what to do + gives a bit more info about environments, so they are not left wondering too much.